v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubStreamLT.java

```diff
@@ -0,0 +1,171 @@
+  private static final String ARTIFACT_BUCKET = TestProperties.artifactBucket();
+  private static final String TEST_ROOT_DIR =
+      TextToPubsubStreamLT.class.getSimpleName().toLowerCase();
+  private static final String INPUT_PATH = "TextToPubsubStream/input.txt";
+  private static final String INPUT_PCOLLECTION =
+      "Read Text Data/Via ReadFiles/Read all via FileBasedSource/Read ranges.out0";
+
+  private static PubsubResourceManager pubsubResourceManager;
+  private static ArtifactClient gcsClient;
+  private static String generatorSchemaPath;
+
+  @Before
+  public void setup() throws IOException {
+    // Set up resource managers
+    pubsubResourceManager =
+        DefaultPubsubResourceManager.builder(testName, project)
+            .credentialsProvider(CREDENTIALS_PROVIDER)
+            .build();
+
+    Storage storageClient = createStorageClient(CREDENTIALS);
+
+    gcsClient = GcsArtifactClient.builder(storageClient, ARTIFACT_BUCKET, TEST_ROOT_DIR).build();
+    // upload schema files and save path
+    generatorSchemaPath =
+        getFullGcsPath(
+            ARTIFACT_BUCKET,
+            gcsClient
+                .uploadArtifact(
+                    "schema.json",
+                    Resources.getResource("TextToPubsubStream/schema.json").getPath())
+                .name());
+
+    // upload input.txt
+    gcsClient.uploadArtifact("input.txt", Resources.getResource(INPUT_PATH).getPath());
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, gcsClient);
+  }
+
+  @Test
+  public void test1Backlog10Mb() throws IOException {
+    testBacklog(Function.identity(), "520", 3, 20);
+  }
+
+  @Test
+  public void test2Backlog10Gb() throws IOException {
+    testBacklog(Function.identity(), "250000", 5, 50);
+  }
+
+  @Test
+  public void test3Backlog100Gb() throws IOException {
+    testBacklog(Function.identity(), "1000000", 20, 60);
+  }
+
+  public void testBacklog(
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder,
+      String qps,
+      long generatorTimeOut,
+      long pipelineTimeOut)
+      throws IOException {
+    // Creates a topic on Pub/Sub.
+    TopicName outputTopic = pubsubResourceManager.createTopic("topic");
+    // Creates a subscription on Pub/Sub.
+    SubscriptionName outputSubscription =
+        pubsubResourceManager.createSubscription(outputTopic, "output-subscription");
+
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaLocation(testName, generatorSchemaPath)
+            .setQPS(qps)
+            .setSinkType("GCS")
+            .setOutputDirectory(getTestMethodDirPath())
+            .setNumShards("20")
+            .setNumWorkers("20")
+            .setMaxNumWorkers("50")
+            .build();
+
+    // Executes the data generator and return approximate number of messages
+    long expectedMessages =
+        (long) (dataGenerator.execute(Duration.ofMinutes(generatorTimeOut)) * 0.99);
```

Comment:
please add a comment about the 0.99 multiplier

Reply:
It's done.